### ISSUE TITLE
Fix Style/MethodCallWithArgsParentheses inconsistency with hash punning

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#10317]
## Expected behavior

With Ruby 3.1.0's new hash punning syntax, if you try to use punning on *more than one line without parentheses*, you'll get a Rubocop error. Here's an inelegant and contrived example where the problem is with the use of parentheses in the constructor:

``` ruby 
Example = Struct.new :one, :two, keyword_init: true

class ExampleWrapper
  def initialize one:, two:
    @first = Example.new(one:, two:)
    @second = Example.new(one:, two:)
  end
end

puts ExampleWrapper.new(one: 1, two: 2).inspect
```

I would expect the above to not throw an error and be valid code since the Ruby parser requires it.

## Actual behavior

When you run the above code as:

``` 
rubocop snippet.rb
```

...you'll get the following errors:

``` 
Offenses:

/Users/bkuhlmann/Engineering/Misc/snippet.rb:19:25: C: [Correctable] Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
    @first = Example.new(one:, two:)
                        ^^^^^^^^^^^^
/Users/bkuhlmann/Engineering/Misc/snippet.rb:20:26: C: [Correctable] Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
    @second = Example.new(one:, two:)
                         ^^^^^^^^^^^^
```

Should you allow Rubocop to auto-correct the above code, you'll end up with the following implementation (truncated for brevity):

``` ruby
  def initialize one:, two:
    @first = Example.new one:, two:
    @second = Example.new one:, two:
  end
```

Without the parenthesis, you'll then have `Lint/Syntax` errors which you can confirm when attempting to run the code snippet:

``` 
/Users/bkuhlmann/Engineering/Misc/snippet.rb:20: syntax error, unexpected label, expecting `end'
    @second = Example.new one:, two:
/Users/bkuhlmann/Engineering/Misc/snippet.rb:24: syntax error, unexpected end-of-input, expecting `end'
...per.new(one: 1, two: 2).inspect
```

The problem appears when you are using hash punning on more than one line in your implementation. If you were to write code like this:

``` ruby
class ExampleWrapper
  def initialize one:, two:
    @first = Example.new one:, two:
  end
end
```

This would be syntactically correct for *both* Rubocop and Ruby. You only need the extra parentheses when using punning on more than one line. Which seems weird for Ruby too? Should this be logged as in issue with Ruby core as well? 

## Steps to reproduce the problem

1. Save the above code snippet as `snippet.rb`.
2. Enable the `Style/MethodCallWithArgsParentheses` cop.
3. Run as `rubocop snippet.rb`.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.0, running on ruby 3.1.0 arm64-darwin21.2.0)
  - rubocop-performance 1.13.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.6.0
```

**Comment by koic** [comment]
> Which seems weird for Ruby too? Should this be logged as in issue with Ruby core as well?

This has been reported to Ruby core and it's an expected spec at least in Ruby 3.0.
https://bugs.ruby-lang.org/issues/18396

I'll take a look at the RuboCop side.

**Comment by bkuhlmann** [comment]
:bow: Koichi. Ah, I should have searched the Ruby core tickets to double check. Thanks, this is good to know!

**Comment by bkuhlmann** [comment]
Hey Koichi. :wave: I'm afraid 1.24.1 didn't resolve this issue. I'm still seeing the following the following errors:

``` 
[Correctable] Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments. (https://rubystyle.guide#method-invocation-parens)
```

Here's the configuration I'm using:

``` 
Style/MethodCallWithArgsParentheses:
  Enabled: true
  EnforcedStyle: omit_parentheses
  AllowParenthesesInChaining: true
  AllowParenthesesInMultilineCall: true
  IgnoreMacros: false
```

...and here's the Rubocop details:

``` 
1.24.1 (using Parser 3.0.3.2, rubocop-ast 1.15.1, running on ruby 3.1.0 arm64-darwin21.2.0)
  - rubocop-performance 1.13.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.7.0
```

Is it the `EnforcedStyle: omit_parentheses` setting that is tripping this up? I realize now I didn't specify that in the original report so maybe that's the root cause?

**Comment by koic** [comment]
Hi, Brooke. I have confirmed that the issue has not been resolved 💦  I will work on this in the few days. Please wait a moment. Thank you!

**Comment by koic** [comment]
I've opened https://github.com/rubocop/rubocop/pull/10346 to resolve the issue again.

**Comment by bkuhlmann** [comment]
🙇 Awesome. OK, I'll follow along there.
